openedx/core/djangoapps/schedules/management/commands/tests/send_email_base.py

```diff
@@ -169,13 +175,27 @@ def test_schedule_bin(self, schedule_count, mock_metric, mock_ace):
             self.assertEqual(mock_schedule_send.apply_async.call_count, schedule_count)
             self.assertFalse(mock_ace.send.called)
 
+    def _schedule_factory(self, offset=None, **factory_kwargs):
+        _, _, target_day, upgrade_deadline = self._get_dates(offset=offset)
```

Comment:
Since almost all the tests that call `_schedule_factory` have already called `_get_dates` themselves, does it make sense to have them pass in the already computed values for `target_day` and `upgrade_deadline`?